_posts/2019-05-28-experience-and-opinions-developing-for-google-assistant-and-amazon-alexa.md

```diff
@@ -0,0 +1,92 @@
+
+[Dialogflow](https://dialogflow.com/) is Google's machine learning natural language processing Alexa equivalent. Currently there is an API for Dialogflow but no supported clients. Everything you want to do has to go through the Dialogflow web console. The models are shared as a zip file that you have to export and import. The content within them is complicated and risky to edit by hand. I did not like having to import a zip file to look at how a demo was implemented.
+
+### Training the Model
+{: .text-left }
+
+To train Alexa to understand what to do when questions were asked was a bit cumbersome. I created a [website](https://agrc.github.io/digital-assisant-skills/) to generate Alexa utterances. Basically all of the ways a user could phrase a question. Dialogflow seemed a lot smarter and required less training. It certainly did not require a website to generate phrases. Google is winning the natural language processing in my opinion and was much easier to train.
+
+### Beyond Voice
+{: .text-left }
+
+Digital assistants lasted without screens for barely a year. Skills can now play videos, show images, and present content in basic HTML components to compliment the voice responses. Alexa has a presentation language [APL](https://developer.amazon.com/docs/alexa-presentation-language/apl-overview.html) to design for this but had limited SDK support. The Google Actions SDK for nodejs had cards, buttons, tables, images, etc all built in. It was simple and fun to add the visual components while developing. The simulator displays everything nicely.
+
+## Testing
+{: .text-left }
+
+I really like the ask CLI way of testing a skill. Talking to your skill over and over again is bad and typing it over and over is worse. The ask CLI supports simple generated [text file recordings](https://github.com/agrc/digital-assisant-skills/blob/alexa-skill/recordings/all.template) that direct the conversation in the test. The CLI outputs the responses and you can visually approve the response. I could not find a way to do this in the Google Assistant simulator. The node sdk does have a concept call `Suggestions` that allow you to click a suggestion bubble instead of typing and that was an improvement but not as nice as the recordings.
+
+### Location
+{: .text-left }
+
+The Google Assistant and Alexa simulators are very nice. They give you screens to test presentation markup as well as other features. One feature that the Alexa simulator is missing is location. You cannot set a location for the simulator to use. Google makes this very simple and it is on the main UI. A major difference between the two assistants is that addresses are assigned to devices in Amazon. Amazon is thinking, where do I ship you all of your Holiday packages. The Google Assistant is on the every Android phone so location is where am I now.
```

Comment:
```suggestion
The Google Assistant and Alexa simulators are very nice. They give you screens to test presentation markup as well as other features. One feature that the Alexa simulator is missing is location. You cannot set a location for the simulator to use. Google makes this very simple and the location is on the main UI. A major difference between the two assistants is that addresses are assigned to devices in Amazon. Amazon is thinking, "Where do I ship you all of your holiday packages?" while the Google Assistant is on every Android phone, so location is "Where am I now?"
```

Reply:
Also, it seems like "nice" is used several times in the article? May be worthwhile to consider other terms.